mav_msgs/include/mav_msgs/common.h

```diff
@@ -158,11 +158,50 @@ inline Eigen::Vector3d omegaFromRotationVector(
                 -rot_vec(1), rot_vec(0), 0.0f;
   // Set up matrix to calculate omega
   Eigen::Matrix3d W;
-  W = Eigen::MatrixXd::Identity(3,3) - phi_matrix * (1-cos(phi))/phi/phi
+  W = Eigen::MatrixXd::Identity(3,3) + phi_matrix * (1-cos(phi))/phi/phi
       + phi_matrix*phi_matrix * (phi-sin(phi))/phi/phi/phi;
   return W * rot_vec_vel;
 }
 
+// Calculates angular acceleration (omegaDot) from rotation vector derivative
+// based on formula derived in "Finite rotations and angular velocity" by Asher Peres
+inline Eigen::Vector3d omegaDotFromRotationVector(
+                            const Eigen::Vector3d& rot_vec,
+                            const Eigen::Vector3d& rot_vec_vel,
+                            const Eigen::Vector3d& rot_vec_acc) {
+  double phi        = rot_vec.norm();
+  double phi_dot    = rot_vec.dot(rot_vec_vel)/phi; 
+
+  if (abs(phi) < 1.0e-3) {
+    // This captures the case of zero rotation
+    return rot_vec_acc;
+  }
+  // Create skew-symmetric matrix from rotation vector and velocity
+  Eigen::Matrix3d phi_matrix;
+  phi_matrix        << 0.0f, -rot_vec(2), rot_vec(1),
```

Comment:
Please define a "skewSymmetricMatrix" function so you don't have to hardcode it every time.